docs/Developers/Design/CRD_design.md

```diff
@@ -0,0 +1,438 @@
+This document describes the set of [Custom Resource Definitions
+(CRDs)](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/)
+that will be used to configure a GCS Gluster cluster. The actual implementation
+of the resources described here will be phased in during development. The
+purpose of this document is to provide the overall structure, ensuring the end
+result provides necessary configurability in a user-friendly manner.
+
+# Overview
+
+A single Gluster operator may control one or more individual Gluster clusters.
+Each cluster could be either hosted within the Kubernetes cluster as a set of
+pods (a converged configuration) or the Gluster nodes could be running on hosts
+outside the Kubernetes cluster (referred to as independent mode). The
+capabilities of the operator differ significantly between these two modes of
+deployment, but the same set of CRDs should be used for both where possible.
+
+A given Gluster cluster is defined by several different Custom Resources (CRs)
+that form a hierarchy. At the top level is the "cluster" CR that describes
+cluster-wide configuration options such as the "name" for the cluster, the TLS
+credentials that will be used for securing communication, and peer clusters for
+geo-replication.
```

Comment:
`peer` cluster -> Does it mean a seperate TSP ?

Reply:
"Peer clusters"... yes separate TSPs, usable as a georep target.

The "peering" for georep is handled at the cluster-level just as georep is set up at the cluster-level today for standalone. Once the peering is done, individual volumes can be replicated. This would be configured via StorageClass parameters that would list the name of this peer cluster and maybe some other options for the volume-level configuration of the replication.